drivers/modem/modem_cmd_handler.c

```diff
@@ -266,13 +275,14 @@ static void cmd_handler_process(struct modem_cmd_handler *cmd_handler,
 		}
 
 		/* load match_buf with content up to the next CR/LF */
+		/* NOTE: keep room in match_buf for ending NUL char */
 		match_len = net_buf_linearize(data->match_buf,
-					      data->match_buf_len,
+					      data->match_buf_len - 1,
 					      data->rx_buf, 0, len);
-		if (data->match_buf_len < match_len) {
+		if (data->match_buf_len - 1 < match_len) {
```

Comment:
Not really an issue but it would be nice to the reader if () would be used here, so `if ((data->match_buf_len - 1) < match_len) {`

Reply:
I'll add parens.